### PR TITLE
simpler UI for one geometry at a time, from APS 29-ID

### LINKED ISSUE #63: provide SPEC-compatibility functions from APS 29-ID
Various functions are provided in the [APS 29-ID](https://git.aps.anl.gov/rodolakis/Macros_29id/-/blob/master/ScanFunctions_hkl.py) (restricted-access) repository that make it easier for SPEC users to remember the new tools in Bluesky's *hklpy* package.

Such as:

```python
def calcUB(eV,r1,r2):
    """Compute the UB matrix with two reflections for a given energy."""
    fourc.calc.wavelength = eV2Lambda(eV)
    fourc.calc.sample.compute_UB(r1, r2)
    print(fourc.calc.sample.UB)

def setor(h,k,l,tth0,omega0,chi0,phi0):
    """Define a reflection in fourc"""
    r1 = fourc.calc.sample.add_reflection(     
        h,k,l,
        position=fourc.calc.Position(tth=tth0, omega=omega0, chi=chi0, phi=phi0))
    return(r1)
```

Here are some of the functions:

function | docstring
--- | ---
`cahkl` | Calculate omega,chi,phi,tth,kth,kap,kphi for a given h,k,l reflection
`listSamples` | List all samples currently defined in fourc; specify  current one.
`newSample` | Add new sample to  diffractometer.
`scanHKL_E` | Uses bluesky pyhkl to calculate the motor positions for an energy scan at constant Q
`setor` | Define a reflection in fourc

Investigate these functions and generalize, yet simplify, for SPEC users.

Thanks, @rodolakis!

### LINKED ISSUE #118: CI coverage does not look at this project
Checking unit test coverage in latest [actions workflow log](https://github.com/bluesky/hklpy/runs/2381935465?check_suite_focus=true#step:8:69), the coverage report is for *ophyd* and not this project.

Will address this in PR #117.  Please wait until this is resolved.

### PR DESCRIPTION
**Pull request by prjemian**
fix #63, fix #118

### CODE REVIEW

**Comment by prjemian** [comment]
@mrakitin @rodolakis @ambarb @gfabbris : ready for review

**Comment by prjemian** [comment]
@rodolakis : You were asking about a quick reference sheet for *hklpy*.  Now, there is a [quick reference sheet](https://github.com/bluesky/hklpy/blob/63-user-2021-04/docs/source/user.rst#user) included with the pull request.

**Comment by prjemian** [comment]
Do not merge until #118 is resolved.

**Comment by prjemian** [comment]
@mrakitin ready, again (code coverage by unit tests is 87%)

File: hkl/user.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+from hkl.util import Lattice
+import numpy
+import pyRestTable
+
+
+_geom_ = None  # selected diffractometer geometry
+
+
+def _check_geom_selected_(*args, **kwargs):
+    """Raise ValueError i no diffractometer geometry is selected."""
```
Comment (mrakitin) [inline hkl/user.py:58]:
```suggestion
    """Raise ValueError if no diffractometer geometry is selected."""
```

File: hkl/user.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+from hkl.diffract import Diffractometer
+from hkl.util import Lattice
+import numpy
+import pyRestTable
+
+
+_geom_ = None  # selected diffractometer geometry
+
+
+def _check_geom_selected_(*args, **kwargs):
```
Comment (mrakitin) [inline hkl/user.py:57]: Was the intent to make this function "private"? I think the trailing underscore is not needed then.

**Review by mrakitin — COMMENTED** [review]
Looks good, thanks @prjemian! I have a couple of minor suggestions.

**Comment by prjemian** [comment]
@mrakitin : Thanks for the review!

**Review by mrakitin — APPROVED** [review]

**Pull request merged**